lua/lspconfig/server_configurations/elmls.lua

```diff
@@ -10,7 +10,7 @@ return {
     filetypes = { 'elm' },
     root_dir = function(fname)
       local filetype = api.nvim_buf_get_option(0, 'filetype')
-      if filetype == 'elm' or (filetype == 'json' and fname:match 'elm%.json$') then
+      if util.ft_matches(filetype, 'elm') or (util.ft_matches(filetype, 'json') and fname:match 'elm%.json$') then
```

Comment:
why can't we use vim.filetype.match with the `fname`, here? (note: the parens in the code below are important because match() has multi-value return)

    if 'elm' == (vim.filetype.match({filename=name}))

Reply:
Been a while since I looked at this code, but I think that should work. I just didn't want to change the semantics of the existing code. e.g. your suggested check wouldn't match files with a manually set `filetype`; although I'm not sure if that is a negligible difference or not. Would lspconfig even spawn a server for e.g. `foobar.txt` where the user calls `:set filetype=elm`?

Reply:
see https://github.com/neovim/nvim-lspconfig/pull/3145#issuecomment-2392302686